### PR TITLE
Change source for individual Beneficiary Child Page to ChildHealthMon…

### PR DESCRIPTION
**Pull request by dmydlo**
…thly table from view
Hi @emord,
I have change source of individual Beneficiary Child Page to a ChildHealthMonthly table from ChildHealthMonthly view to make it respond faster.
Let me know if You have any questions.
Regards, Dawid.

### CODE REVIEW

**Review by emord — CHANGES_REQUESTED** [review]
I think that this won't help. The issue is that this query is doing some sequential scans on some old tables that only have an index on (awc_id, case_id). I added these indexes to the old tables, but I think it is still good practice to change it to awc_id, case_id

Can you change this to also filter using the `awc_id` ? It should be possible to pass that in to this function from the view

**Comment by dmydlo** [comment]
Hi @emord @lwyszomi,
I have updated PR with filtering by awc_id.

Regards, Dawid

File: custom/icds_reports/reports/awc_reports.py [hunk 1]
```diff
@@ -1040,10 +1040,11 @@ def base_data(row_data):
     return config
 
 
-@quickcache(['case_id'], timeout=30 * 60)
-def get_beneficiary_details(case_id):
-    data = ChildHealthMonthlyView.objects.filter(
-        case_id=case_id
+@quickcache(['case_id', 'awc_id'], timeout=30 * 60)
+def get_beneficiary_details(case_id, awc_id):
+    data = ChildHealthMonthly.objects.filter(
```
Comment (emord) [inline custom/icds_reports/reports/awc_reports.py]: I think that you can move this back to `ChildHealthMonthlyView` as that wasn't the cause for slowness

**Review by emord — APPROVED** [review]
Thanks!

File: custom/icds_reports/reports/awc_reports.py [hunk 2]
```diff
@@ -1040,10 +1040,11 @@ def base_data(row_data):
...
 
 
-@quickcache(['case_id'], timeout=30 * 60)
-def get_beneficiary_details(case_id):
+@quickcache(['case_id', 'awc_id'], timeout=30 * 60)
+def get_beneficiary_details(case_id, awc_id):
     data = ChildHealthMonthlyView.objects.filter(
-        case_id=case_id
+        case_id=case_id,
+        awc_ic=awc_id
```
Comment (emord) [inline custom/icds_reports/reports/awc_reports.py]: `awc_ic`  -> `awc_id`

**Pull request merged**